test/functional/readpreference.test.js

```diff
@@ -759,4 +759,35 @@ describe('ReadPreference', function() {
         });
     })
   });
+
+  it('should use session readPreference instead of client readPreference', {
+    metadata: { requires: { topology: ['single', 'replicaset'] } },
+    test: function(done) {
+      const configuration = this.configuration;
+      const client = this.configuration.newClient(configuration.writeConcernMax(), {
+        useUnifiedTopology: true,
+        readPreference: 'primaryPreferred'
+      });
+
+      client.connect((err, client) => {
+        this.defer(() => {
+          client.close();
+        });
+        expect(err).to.be.null;
+        expect(client).to.not.be.null;
```

Comment:
```suggestion
        expect(err).to.not.exist;
        expect(client).to.exist;
```
These "exists" expectations are safer than checking against null/undefined when you don't care about distinguishing them.